modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java

```diff
@@ -133,20 +138,28 @@ private void populateSource(Source source, OMElement sourceEle) {
     private void populateTarget(Target target, OMElement sourceEle) {
         // type attribute
         OMAttribute typeAttr = sourceEle.getAttribute(ATT_TYPE);
+        OMAttribute actionAttr = sourceEle.getAttribute(ATT_ACTION);
+
+        if (actionAttr != null && actionAttr.getAttributeValue() != null) {
+            target.setAction(actionAttr.getAttributeValue());
+        } else {
+            target.setAction("replace");
+        }
+
         if (typeAttr != null && typeAttr.getAttributeValue() != null) {
-            int type = convertTypeToInit(typeAttr.getAttributeValue());
+            int type = convertTypeToInt(typeAttr.getAttributeValue());
             if (type >= 0) {
                 target.setTargetType(type);
+                if (type == 1) {
+                    if (!target.getAction().equals("replace")) {
```

Comment:
we may have to change this depending on the result of the test I have mentioned below comment on (sourceType = custom targetType = envelope and action=child)

Reply:
when we have targetType = envelope, then the source value should be an envelope. Otherwise it should show the following error. "ERROR - EnrichMediator SOAPEnvelope is expected"
If we have targetType = envelope and an source value also envelope, then the output will be an envelope within an envelope. So this is not a valid combination.

Reply:
ok, thanks.